mautic/mautic
# Javascript handling for DynamicContent

Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | N
| New feature?  | Y
| BC breaks?    | N
| Deprecations? | N
| Fixed issues  |  N/A

## Description

Adds the ability to use javascript to embed dynamic content in any website that loads the mautic js.

## Steps to reproduce the bug (if applicable)

## Steps to test this PR

See testing instructions on #1882. Once you've completed that, you can navigate to any website where you've embedded the `mtc.js` from the mautic install where you've merged this PR for testing. Then, on that page, place the following, replacing `slot_name` with the name of the slot you created in your campaign in the earlier testing instructions.

```html
<div class="mautic-slot" data-slot-name="slot_name">
Some default text here
</div>
```

I've also added the ability to secure your incoming CORS requests. A new section on the System Settings configuration tab allows you to decide whether or not to restrict which domains may make CORS requests to your Mautic installation. Look for the "CORS Settings" block, and the first option you'll see is a Yes/No toggle. Select "No" to make CORS requests available to all domains. If you'd like to secure which domains may make CORS requests, select "Yes" and then enter the valid fully qualified domains (`http://yourdomain.com`), without a trailing slash, one per line.
